pkg/terminal/command.go

```diff
@@ -1667,14 +1668,16 @@ func examineMemoryCmd(t *Term, ctx callContext, args string) error {
 				return fmt.Errorf("size must be a positive integer (<=8)")
 			}
 		default:
-			if i != len(v)-1 {
-				return fmt.Errorf("unknown option %q", v[i])
+			expr := strings.Join(v[i:], " ")
```

Comment:
You can't do it this way because `examinemem m["one   two"]` won't work. You have to do something like this:

```
const (
    fmtPrefix = "-fmt "
    countPrefix = "-count "
)
for args != "" {
   switch {
   case strings.HasPrefix(args, fmtPrefix):
       args = args[len(fmtPrefix):...]
       // ... 
   case strings.HasPrefix(args, countPrefix):
       args = args[len(countPrefix):....]
       // ...
   default:
       // eval expression in args
      args = ""
   }
}
```

Also you should update the documentation of `examinemem` above.

Reply:
On second thought it's probably simpler and better to have an explicit option `-x` that will interpret the rest of the args string as an expression. Still you will have to parse one argument at a time `strings.SplitN(args, " ", 2)` instead of just calling `strings.FieldFunc` like it does now.

Reply:
Tried my best in 49cf77ca0f16909ad2c152ef8a4471c9e2ec4de0 - let me know what ya think.

Reply:
The way `-x` is implemented is fine but there's still the problem that the entire argument string is split in advance and `examinemem m["one  two"]` with two spaces between `one` and `two` will be mangled.

Reply:
I'm trying to understand and reproduce the two-space issue you're describing. But I'm unable to see it.

Here is a quick test against `dlv` compiled from my local branch: https://gist.github.com/felixge/b4be92dbca9fbaba1a62488b2cdb9a83

Reply:
You are right, with `strings.Split` (instead of strings.FieldsFunc) that problem doesn't exist but there's another problem:

```
(dlv) examinemem -fmt oct  -x &s1
```

(two spaces between `oct` and `-x`) fails with `Command failed: unknown option ""`

Reply:
PS. the problem is: spaces shouldn't be significant before `-x` but can be significant after `-x`.

Reply:
@aarzilli ah, got it! I think it all makes sense now, I'll update the patch to deal with this.

Reply:
3e16767894c3e70f203eeab071e1a43858b53928 should take care of double space handling. PTAL